### PR TITLE
ISSUE-283:Facet Summary Processor misbehavior on Date Range when it is the only Facet selected

### PR DESCRIPTION
**Pull request by DiegoPino**
See #283 

This one was tricky. Gist: To generate a "unselect the current facet" you have to build all active facets except the one you want to unselect. When there are no ones (means the current, in this case our custom range) Facet Module suggests (copied the code) to use a HTTP REQUEST based parser method that gets confused when a single View (Facet Source) has multiple Displays (specially if those display are not 1:1).

The solution was not terrible.

Now that this is working I also made "quoting" the Full Text Search Terms Facet Summary (part of this pull)

### CODE REVIEW

**Comment by DiegoPino** [comment]
@aksm if you have a few minutes, could you check if you see some non-sense here? Code is tested but maybe I missed a comment, indentation, etc or a piece of the logic is bonkers! Thanks

File: modules/format_strawberryfield_facets/config/schema/format_strawberryfield_facets.schema.yml [hunk 1]
```diff
@@ -64,13 +75,16 @@ plugin.plugin_configuration.facets_summary_processor.sbf_last_active_facets:
...
       type: integer
-      label: Enable Empty Message
+      label: 'Enable Empty Message'
     enable_query:
       type: integer
-      label: Enable Full text search terms as Facet Summary Entry
+      label: 'Enable Full text search terms as Facet Summary Entry'
+    quote_query:
+      type: boolean
+      label: 'If Fulll Text search terms will be surrounded by Double Quotes or not'
```
Comment (aksm) [inline modules/format_strawberryfield_facets/config/schema/format_strawberryfield_facets.schema.yml:87]:
```suggestion
      label: 'If Full Text search terms will be surrounded by Double Quotes or not'
```

File: modules/format_strawberryfield_facets/src/Plugin/facets/widget/DateRangeWidget.php [hunk 2]
```diff
@@ -227,7 +227,7 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
     ];
     $form += parent::buildConfigurationForm($form, $form_state, $facet);
 
-    $message = $this->t('To achieve the standard behavior of a Format Strawberryfield Date Range Picker, you need to enable the facet setting below <em>"Format Strawberryfield Date Range Picker"</em>.');
+    $message = $this->t('The Format Strawberryfield Date Range Picker Widget requires you to check the facet setting below <em>"Format Strawberryfield Date Range Picker" and inside it check "enabled" it.</em>.');
```
Comment (aksm) [inline modules/format_strawberryfield_facets/src/Plugin/facets/widget/DateRangeWidget.php:230]:
```suggestion
    $message = $this->t('The Format Strawberryfield Date Range Picker Widget requires you to check the facet setting below <em>"Format Strawberryfield Date Range Picker" and check "enabled" inside it.</em>.');
```

**Review by aksm — CHANGES_REQUESTED** [review]

**Comment by DiegoPino** [comment]
@aksm thanks! I committed your change requests/suggestions

**Pull request merged**